llvm/lib/Transforms/IPO/FunctionAttrs.cpp

```diff
@@ -126,8 +126,21 @@ static void addLocAccess(MemoryEffects &ME, const MemoryLocation &Loc,
     return;
   if (isa<Argument>(UO)) {
     ME |= MemoryEffects::argMemOnly(MR);
+    if (Loc.Size <= LocationSize::precise(sizeof(int)))
+      ME |= MemoryEffects::errnoMemOnly(MR);
     return;
   }
+  if (isa<CallInst>(UO)) {
+    static constexpr auto ErrnoFnNames = {"__errno_location", "_errno",
+                                          "__errno", "___errno"};
+    auto *Callee = cast<CallInst>(UO)->getCalledFunction();
+    if (Callee && llvm::any_of(ErrnoFnNames, [&](const auto Fn) {
+          return Fn == Callee->getName();
+        })) {
+      ME |= MemoryEffects::errnoMemOnly(MR);
+      return;
+    }
+  }
```

Comment:
Let's please not introduce any special inference logic for errno in this patch, we can do this in a followup.

Reply:
Note that the bit you actually *want* to modify the the code a few lines below that uses IRMemLocation::Other. That one should not include Errno as well.

Reply:
>  That one should not include Errno as well

You meant that we should include the memory effects of Errno, as Other doesn't affect Errno, correct?

Reply:
Yes